app/controllers/api/v2/providers_controller.rb

```diff
@@ -103,21 +104,61 @@ def get_user
         @user = User.find(params[:user_id])
       end
 
+      def update_accrediting_enrichment
+        return if accredited_bodies_params.values.none?
+
+        enrichment = @provider.enrichments.find_or_initialize_draft
+
+        enrichment.accrediting_provider_enrichments =
+          accredited_bodies_params["accredited_bodies"].map do |accredited_body|
+            {
+              UcasProviderCode: accredited_body["provider_code"],
+              Description: accredited_body["description"],
+            }
+          end
+
+        set_provider_code enrichment
+        enrichment.save
+      end
+
       def update_enrichment
         return unless enrichment_params.values.any?
 
         enrichment = @provider.enrichments.find_or_initialize_draft
         enrichment.assign_attributes(enrichment_params)
 
+        set_provider_code enrichment
+        enrichment.save
+      end
+
+      def set_provider_code(enrichment)
```

Comment:
Can we have this as an `after_create` hook on the model? Can be a new ticket, of course.

Reply:
https://trello.com/c/3KknUZq1/1903-refactor-move-setprovidercodeenrichment-to-hook